blocks/monitoring/evaluators.py

```diff
@@ -211,8 +248,23 @@ def process_batch(self, batch):
         if self._accumulate_fun is not None:
             self._accumulate_fun(**batch)
 
+        try:
+            batch = dict_subset(batch, self.buffer2_.input_names)
+        except KeyError:
+            reraise_as(
+                "Not all data sources required for monitoring were"
+                " provided. The list of required data sources:"
+                " {}.".format(self.buffer2_.input_names))
+        if self._accumulate_fun2 is not None:
+            self.non_theano_variables[-1].accumulate(self._accumulate_fun2(**batch)[0], self._accumulate_fun2(**batch)[1])
```

Comment:
Here I should iterate over variables and call their ```accumulate```. Right?
I'm still not sure how to keep track of ```requires``` for each ```MonitoredQuantity```. @bartvm , could you let me know where should I create that dictionary which maps from variable to its requires?